### PR TITLE
Change CFT vnet to cft-${var.env}-vnet rather than core-infra-vnet-${var.env}

### PR DESCRIPTION
**Pull request by thomast1906**
### Change description ###
Change CFT vnet to cft-${var.env}-vnet rather than core-infra-vnet-${var.env}
- Will match same as SDS

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by kunlebalogun — APPROVED** [review]

**Review by adusumillipraveen — APPROVED** [review]

**Comment by thomast1906** [comment]
@timja  was following https://github.com/hmcts/cnp-module-api-mgmt-private/blob/main/init.tf 

may have been same creator.. :D

**Comment by timja** [comment]
That module isn't for users it's a one-off module for creating it, bit different

File: README.md [hunk 1]
```diff
@@ -41,6 +41,7 @@ module "postgresql" {
 |------|---------|
 | <a name="provider_azuread"></a> [azuread](#provider\_azuread) | n/a |
 | <a name="provider_azurerm"></a> [azurerm](#provider\_azurerm) | >= 3.7.0 |
+| <a name="provider_azurerm.aks_subscription"></a> [azurerm.aks\_subscription](#provider\_azurerm.aks\_subscription) | >= 3.7.0 |
```
Comment (timja) [inline README.md:44]: maybe `postgres_network`?

it's nothing to do with aks it's the provider that has the subscription that the subnet is in.

File: pgsql-flexible-server.tf [hunk 2]
```diff
@@ -4,8 +4,8 @@ locals {
   server_name            = "${local.name}-${var.env}"
   postgresql_rg_name     = var.resource_group_name == null ? azurerm_resource_group.rg[0].name : var.resource_group_name
   postgresql_rg_location = var.resource_group_name == null ? azurerm_resource_group.rg[0].location : var.location
-  vnet_rg_name           = var.business_area == "sds" ? "ss-${var.env}-network-rg" : "core-infra-${var.env}"
-  vnet_name              = var.business_area == "sds" ? "ss-${var.env}-vnet" : "core-infra-vnet-${var.env}"
+  vnet_rg_name           = var.business_area == "sds" ? "ss-${var.env}-network-rg" : "cft-${var.env}-network-rg"
+  vnet_name              = var.business_area == "sds" ? "ss-${var.env}-vnet" : "cft-${var.env}-vnet"
```
Comment (timja) [inline pgsql-flexible-server.tf:8]: impact of changing this on places like draftstore? or anywhere else?

**Review by timja — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (thomast1906) [reply README.md:44]: yeah, I dont mind - was just copying similar to Jenkins, updated.

File: README.md [hunk 3]
```diff
@@ -4,7 +4,19 @@ Terraform module for [Azure Database for PostgreSQL - Flexible Server](https://d
...
+  alias                      = "postgres_network"
+  subscription_id            = var.aks_subscription_id
+}
+
 module "postgresql" {
+
+  providers = {
+    azurerm.postgres_network = azurerm.postgres_network
+  }
+
```
Comment (timja) [inline README.md:19]: should only be needed if you are renaming the provider I think
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -4,7 +4,19 @@ Terraform module for [Azure Database for PostgreSQL - Flexible Server](https://d
 ## Example
 
 ```hcl
+provider "azurerm" {
+  features {}
+  skip_provider_registration = true
+  alias                      = "postgres_network"
+  subscription_id            = var.aks_subscription_id
+}
```
Comment (timja) [inline README.md:12]: is it worth creating separate code blocks for this do you think?

e.g.

provider.tf
```terraform
provider "azurerm" {
  features {}
  skip_provider_registration = true
  alias                      = "postgres_network"
  subscription_id            = var.aks_subscription_id
}
```

variables.tf
```terraform
variable "aks_subscription_id" {} # provided by the Jenkins library, ADO users will need to specify this
```

postgres.tf
```terraform
...
```

Comment (thomast1906) [reply README.md:12]: yeah, will do that as its beginning to bloat slightly

File: README.md (hunk 3 above)
Comment (thomast1906) [reply README.md:19]: Is this not needed on teams module reference to init with correct provider?

Comment (timja) [reply README.md:19]: i don't think so, all providers should be passed through, you can test it from the example folder

File: README.md [hunk 5]
```diff
@@ -3,8 +3,28 @@ Terraform module for [Azure Database for PostgreSQL - Flexible Server](https://d
...
+
+provider "azurerm" {
+  features {}
+  skip_provider_registration = true
+  alias                      = "postgres_network"
+  subscription_id            = var.aks_subscription_id
+}
+```
+
+main.tf
```
Comment (timja) [inline README.md:20]: ?
```suggestion
postgres.tf
```

File: README.md [hunk 6]
```diff
@@ -3,8 +3,28 @@ Terraform module for [Azure Database for PostgreSQL - Flexible Server](https://d
...
+```hcl
+provider "azurerm" {
+  features {}
+}
+
+provider "azurerm" {
+  features {}
+  skip_provider_registration = true
+  alias                      = "postgres_network"
+  subscription_id            = var.aks_subscription_id
```
Comment (timja) [inline README.md:16]: missing variables.tf for the `aks_subscription_id`

Comment (thomast1906) [reply README.md:16]: some reason didn't push it initially - now added

File: README.md (hunk 3 above)
Comment (thomast1906) [reply README.md:27]: Seems it is needed
```
 The child module requires an additional configuration for provider
│ hashicorp/azurerm, with the local name "azurerm.postgres_network".
```
GH Action: https://github.com/hmcts/terraform-module-postgresql-flexible/actions/runs/4562095063/jobs/8048854878?pr=31

PR: https://github.com/hmcts/terraform-module-postgresql-flexible/pull/31/commits/4497cee6f5735f1ab26a2808f587468be18f61df

**Review by timja — APPROVED** [review]

File: example/variables.tf [hunk 7]
```diff
@@ -1,3 +1,7 @@
 variable "env" {
   default = "test"
 }
+
+variable "aks_subscription_id" {
+  default = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
```
Comment (timja) [inline example/variables.tf:6]: shouldn't this be a valid subscription id? so that this works

**Pull request closed without merging**